packages/itwinui-react/src/core/Tree/TreeNode.tsx

```diff
@@ -230,23 +270,37 @@ export const TreeNode = (props: TreeNodeProps) => {
     >
       {
         <Box
-          className={cx('iui-tree-node', {
-            'iui-active': isSelected,
-            'iui-disabled': isDisabled,
-          })}
+          as='div'
           style={{ '--level': nodeDepth } as React.CSSProperties}
           onClick={() => !isDisabled && onSelected?.(nodeId, !isSelected)}
+          {...nodeProps}
+          className={cx(
+            'iui-tree-node',
+            {
+              'iui-active': isSelected,
+              'iui-disabled': isDisabled,
+            },
+            nodeProps?.className,
+          )}
         >
           {checkbox && (
-            <Box className='iui-tree-node-checkbox'>
+            <Box
+              as='div'
+              {...checkboxProps}
+              className={cx('iui-tree-node-checkbox', checkboxProps?.className)}
+            >
               {React.isValidElement(checkbox)
                 ? React.cloneElement(checkbox as JSX.Element, {
                     tabIndex: isFocused ? 0 : -1,
                   })
                 : checkbox}
             </Box>
           )}
-          <Box className='iui-tree-node-content'>
+          <Box
+            as='div'
+            {...contentProps}
+            className={cx('iui-tree-node-content', contentProps?.className)}
+          >
             {hasSubNodes && expander}
             {hasSubNodes && !expander && (
               <TreeNodeExpander
```

Comment:
Added in this [commit](https://github.com/iTwin/iTwinUI/pull/2022/commits/4d856ed86a6b73969abfe73d92c7a36dac65f411).

Reply:
Why is this one called `treeNodeExpanderProps` when none of the others have a `treeNode` prefix? Should just be `expanderProps`

Reply:
It was named that because it was associated with the `TreeNodeExpander` component. The distinction isn't necessary though, since it is the only expander associated with the `TreeNode` component. Renamed in this [commit](https://github.com/iTwin/iTwinUI/pull/2022/commits/2dacdad18d90076bb2cd5c1898039a8e74eedaf7).